bin/plugin/commands/test-plugins.js

```diff
@@ -0,0 +1,503 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs-extra' );
+const { execSync, spawnSync } = require( 'child_process' );
+
+/**
+ * Internal dependencies
+ */
+const { log, formats } = require( '../lib/logger' );
+
+/**
+ * @typedef WPTestPluginsCommandOptions
+ *
+ * @property {string=} siteType Optional site type, 'single' or 'multi'. Defaults to single.
+ */
+
+/**
+ * @typedef WPDoReplaceWpEnvContent
+ *
+ * @property {string} wpEnvPluginsRegexPattern Regex to match against plugins property of .wp-env.json.
+ * @property {string} wpEnvFile                Path to the plugin tests specific .wp-env.json file.
+ * @property {string} wpEnvDestinationFile     Path to the final base .wp-env.json file.
+ * @property {Object} builtPlugins             Array of plugins slugs from plugins json file.
+ */
+
+/**
+ * @typedef WPDoRunUnitTests
+ *
+ * @property {string} siteType       Site type. 'single' or 'multi'.
+ * @property {Object} builtPlugins   Array of plugin slugs from plugins json file.
+ * @property {Object} disablePlugins Array of plugin slugs to deactivate on test run.
+ * @property {Object} enablePlugins  Array of plugin slugs to activate on test run.
+ */
+
+/**
+ * @typedef WPTestPluginsSettings
+ *
+ * @property {string} pluginsJsonFile       Path to the plugins.json file used for building plugins.
+ * @property {string} siteType              Site type. 'single' or 'multi'.
+ * @property {string} pluginTestAssets      Path to 'plugin-tests' folder.
+ * @property {string} builtPluginsDir       Path to 'build' directory.
+ * @property {string} wpEnvFile             Path to the plugin tests specific .wp-env.json file.
+ * @property {string} wpEnvDestinationFile  Path to the final base .wp-env.json file.
+ * @property {string} performancePluginSlug Slug of the main WPP plugin.
+ */
+
+exports.options = [
+	{
+		argname: '-s, --sitetype <siteType>',
+		description: 'Whether to test "single" (default) or "multi" site.',
+	},
+];
+
+// Switch for flagging start of wp-env so it is not started multiple times.
+let isWpEnvStarted = false;
+
+/**
+ * Command for testing all built standalone plugins.
+ *
+ * @param {WPTestPluginsCommandOptions} opt Command options.
+ */
+exports.handler = async ( opt ) => {
+	doRunStandalonePluginTests( {
+		pluginsJsonFile: './plugins.json', // Path to plugins.json file.
+		siteType: opt.sitetype || 'single', // Site type.
+		pluginTestAssets: './plugin-tests', // plugin test assets.
+		builtPluginsDir: './build/', // Built plugins directory.
+		wpEnvFile: './plugin-tests/.wp-env.json', // Base .wp-env.json file for testing plugins.
+		wpEnvDestinationFile: './.wp-env.override.json', // Destination .wp-env.override.json file at root level.
+		wpEnvPluginsRegexPattern: '"plugins": \\[(.*)\\],', // Regex to match plugins string in .wp-env.json.
+		performancePluginSlug: 'performance', // The plugin slug of the main WPP plugin.
```

Comment:
This must not be hard-coded, it needs to be set based on whichever folder the plugin is in. See a related issue #673 that had a similar problem elsewhere. We should not assume the plugin is in a `performance` folder.

Reply:
@felixarntz addressed in a recent commit, thank you.